Tests/IntegrationTests/Roslyn/RoslynGenTests.cs

```diff
@@ -59,25 +59,26 @@ public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
 
             Assert.IsNotNull(jarInfo, "Failed to process the generated jar successfully");
 
-            AssertExpectedManifestValue(WellKnownPluginProperties.Key, "analyzer1.pkgid1", jarInfo);
+            AssertExpectedManifestValue(WellKnownPluginProperties.Key, "analyzer1pkgid1", jarInfo);
 
             // Check that NuGet package properties have been correctly mapped to the plugin manifest
+            AssertExpectedManifestValue("Plugin-Key", "analyzer1pkgid1", jarInfo); // plugin-key should be lowercase and alphanumeric
             AssertPackagePropertiesInManifest(analyzerPkg, jarInfo);
 
             // Check for the expected property values required by the C# plugin
             // Property name prefixes should be lower case; the case of the value should be the same as the package id
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.analyzerId", "Analyzer1.Pkgid1", jarInfo);
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.ruleNamespace", "Analyzer1.Pkgid1", jarInfo);
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.nuget.packageId", "Analyzer1.Pkgid1", jarInfo);
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.nuget.packageVersion", "1.0.2", jarInfo);
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.staticResourceName", "Analyzer1.Pkgid1.1.0.2.zip", jarInfo);
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.pluginKey", "analyzer1.pkgid1", jarInfo);
-            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.pluginVersion", "1.0.2", jarInfo);
+            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.analyzerId", "Analyzer1.Pkgid1", jarInfo);
+            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.ruleNamespace", "Analyzer1.Pkgid1", jarInfo);
+            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.nuget.packageId", "Analyzer1.Pkgid1", jarInfo);
+            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.nuget.packageVersion", "1.0.2", jarInfo);
+            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.staticResourceName", "Analyzer1.Pkgid1.1.0.2.zip", jarInfo);
+            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.pluginKey", "analyzer1pkgid1", jarInfo);
+            AssertExpectedPropertyDefinitionValue("analyzer1.pkgid1.cs.pluginVersion", "1.0.2", jarInfo);
```

Comment:
I've amended the repository key to include the language as a prefix. This means would could put VB and C# repositories/analyser assemblies in the same jar in the future if we wanted to.